### ISSUE TITLE
Panic on Post

### ISSUE THREAD

**Issue by pastukhov** [loicortola/nodemcu-espress#7]
Hi!
I'm trying to make a relay controller
My relay.post.lua: http://pastebin.com/m3nQ5mZE
and my index.html: http://pastebin.com/3TBJaDsD
I'm getting 

```
Begin Http request: 29536
Connection initialized with id 263
Request callbacks have been set: 29104
PANIC: unprotected error in call to Lua API (bad argument #2 to 'on' (function or lightfunction expected, got nil))
```

Where i'm wrong?

**Comment by loicortola** [comment]
Hi @pastukhov , to help you I will need a little more info if you don't mind:
- What version of the NodeMCU firmware are you using?
- Can you send me the output of file.list() ?

Thank you,

**Comment by pastukhov** [comment]
1.5.1 with today git
I'll send the filelist tomorow morning( i'm GMT+3)
How do you debug the compiled code?

**Comment by pastukhov** [comment]
My code https://github.com/pastukhov/nodemcu-relay.git and list of files

```
auth_api_key.lc : 636 bytes
configNetSmart.lc : 1092 bytes
espress.lc      : 2752 bytes
espress_init.lc : 1504 bytes
http-200        : 12 bytes
http-302        : 26 bytes
http-400        : 20 bytes
http-401        : 21 bytes
http-403        : 18 bytes
http-404        : 19 bytes
http-405        : 27 bytes
http-500        : 30 bytes
http-503        : 28 bytes
http_default_handler.lc : 176 bytes
http_getondata.lc : 460 bytes
http_prototypes.lc : 948 bytes
http_request.lc : 472 bytes
http_request_buffer.lc : 680 bytes
http_request_processor.lc : 668 bytes
http_response.lc : 268 bytes
http_response_send.lc : 584 bytes
http_response_sendfile.lc : 1348 bytes
httpd.lc        : 520 bytes
init.lua        : 756 bytes
router.lc       : 856 bytes
routes/askme.post.lua : 659 bytes
routes/hello.get.lua : 135 bytes
routes/relay.post.lc : 520 bytes
routes/relay.post.lua : 527 bytes
routes_auto.lc  : 880 bytes
static/index.html : 1269 bytes
static/relay.css : 1388 bytes
type-css        : 32 bytes
type-html       : 33 bytes
type-ico        : 21 bytes
type-jpg        : 19 bytes
type-js         : 39 bytes
type-json       : 40 bytes
type-png        : 18 bytes
```

**Comment by pastukhov** [comment]
Same error with echo {"question": "How are you?"} | POST http://192.168.100.198/api/askme

**Comment by loicortola** [comment]
Hi @pastukhov 
I have taken a look at your code, and it's pretty neat. Nothing wrong with it.
I think this may come from the version of the NodeMCU firmware.
As of today, the only firmware I support is the master branch (1.4.0).
I can take a look in the week-end and see which API has changed since, but I think changing your firmware to the master version would solve the problem.

PS: thoughts: don't forget to check that you have all the modules you need (cjson, file, etc...) in your firmware.

**Comment by pastukhov** [comment]
@loicortola Any news?

**Comment by loicortola** [comment]
Hi @pastukhov ,
I haven't taken the time to do it. Easter week-end comes in handy, I will check tomorrow for sure.
Have you tried with a 1.4.0 firmware? (master-branch).
I am pretty sure everything would work with this one.

**Comment by loicortola** [comment]
@pastukhov I successfully reproduced your bug. Will push fix today.

**Comment by loicortola** [comment]
Fixed in 0.4.1 ;)
Let me know how it works!

**Comment by pastukhov** [comment]
Thanks, i'll try tomorow.
